Add model for showing and importing vCalendar and iCalendar events from file

pgerdt opened the pull request:
The contents of a vCalendar or iCalendar file are read to memory, and a model is populated with the event data from the file. Incidences that are not events are discarded. Events can then be imported to a notebook.

pvuorela commented on src/calendarutils.cpp:
```diff
@@ -0,0 +1,193 @@
+    NemoCalendarData::EventOccurrence occurrence;
+    if (event) {
+            QDateTime dtStart = event->dtStart().toLocalZone().dateTime();
+            QDateTime dtEnd = event->dtEnd().toLocalZone().dateTime();
```
Extra indentation.

pvuorela commented on src/calendarimportmodel.cpp:
```diff
@@ -0,0 +1,240 @@
+    mFileName = fileName;
+    emit fileNameChanged();
+    if (!mFileName.isEmpty())
+        importToMemory(fileName);
```
Should setting empty file name also clear current events from memory?

pvuorela commented on src/calendarimportmodel.cpp:
```diff
@@ -0,0 +1,240 @@
+    }
+}
+
+bool NemoCalendarImportModel::importToNotebook(const QString notebookUid)
```
Should probably be const reference.

pvuorela commented on src/calendarimportmodel.cpp:
```diff
@@ -0,0 +1,240 @@
+    for (int i = 0; i < incidenceList.size(); i++) {
+        KCalCore::Incidence::Ptr incidence = incidenceList.at(i);
+        if (incidence->type() == KCalCore::IncidenceBase::TypeEvent)
+            mEventList.append(incidence.staticCast<KCalCore::Event>());
```
To be noted that exceptions, i.e. instances having recurrence id, on recurring events will show up as separate incidences on this model.

pvuorela commented on src/calendarutils.h:
```diff
@@ -0,0 +1,56 @@
+{
+public:
+    NemoCalendarUtils();
+    ~NemoCalendarUtils();
```
As mentioned, the class part is quite unnecessary and namespace would suffice.

pgerdt commented:
Updated PR:
- Changed class NemoCalendarUtils into a namespace
- Clear model when filename is empty (https://github.com/nemomobile/nemo-qml-plugin-calendar/pull/52/files#diff-6a5cb3ed7dc535bdc9481bbd84566208R78)
- Use const reference, indentation fixes

pvuorela commented:
LGTM
